### DISCUSSION TITLE
Is this library using the PHP exec() function to run git command?

### DISCUSSION THREAD

**Discussion by BadPistol97** [czproject/git-php#65] [Q&A] [answered]
The exec() function of PHP is disabled on a shared hosting. I need to use the exec() for Git Hub Web hooks push event on the shared hosting. I need to know if this library use the exec() function.

**Comment by janpecha** [comment]
Hello, yes, this library uses exec() function (and proc_open in some methods).

**Comment by BadPistol97** [comment]
@janpecha is git pull using exec() or proc_open?

**Comment by janpecha** [comment]
@BadPistol97 exec()

**Comment by janpecha** [comment, marked as answer]
Hello, in current version (4.0) it uses only proc_open
